src/Stratis.Bitcoin/P2P/PeerConnector.cs

```diff
@@ -214,11 +214,23 @@ public void StartConnectAsync()
         }
 
         /// <summary>Attempts to connect to a random peer.</summary>
-        internal async Task ConnectAsync(PeerAddress peerAddress)
+        public async Task ConnectAsync(PeerAddress peerAddress)
         {
             if (this.selfEndpointTracker.IsSelf(peerAddress.Endpoint))
             {
-                this.logger.LogTrace("{0} is self. Therefore not connecting.", peerAddress.Endpoint);
+                this.logger.LogTrace("Connect aborted: {0} is self.", peerAddress.Endpoint);
+                return;
+            }
+
+            if (this.IsPeerConnected(peerAddress.Endpoint))
+            {
+                this.logger.LogTrace("Connect aborted: {0} is already connected.", peerAddress.Endpoint);
```

Comment:
We should start to use Debug for manual logs

Reply:
Ah yes will fix 👍